src/main/java/org/openlmis/buq/domain/buq/BottomUpQuantification.java

```diff
@@ -0,0 +1,196 @@
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.javers.core.metamodel.annotation.TypeName;
+import org.openlmis.buq.domain.BaseTimestampedEntity;
+import org.openlmis.buq.dto.buq.BottomUpQuantificationLineItemDto;
+
+@Entity
+@TypeName("BottomUpQuantification")
+@Table(name = "bottom_up_quantifications", schema = "buq")
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class BottomUpQuantification extends BaseTimestampedEntity {
+
+  @NotNull
+  @Getter
+  @Setter
+  private UUID facilityId;
+
+  @NotNull
+  @Getter
+  @Setter
+  private UUID programId;
+
+  @NotNull
+  @Getter
+  @Setter
+  private UUID processingPeriodId;
+
+  @NotNull
+  @Getter
+  @Setter
+  private Integer targetYear;
+
+  @NotNull
+  @Enumerated(EnumType.STRING)
+  @Getter
+  @Setter
+  private BottomUpQuantificationStatus status;
+
+  @OneToMany(
+      mappedBy = "bottomUpQuantification",
+      cascade = {CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH, CascadeType.REMOVE},
+      fetch = FetchType.LAZY,
+      orphanRemoval = true)
+  @Getter
+  @Setter
+  private List<BottomUpQuantificationLineItem> bottomUpQuantificationLineItems;
+
+  @OneToMany(
+      mappedBy = "bottomUpQuantification",
+      cascade = CascadeType.ALL)
+  @Getter
+  @Setter
+  private List<BottomUpQuantificationStatusChange> statusChanges = new ArrayList<>();
+
+  /**
+   * Constructor.
+   *
+   * @param facilityId         id of the Facility
+   * @param programId          id of the Program
+   * @param processingPeriodId id of the ProcessingPeriod
+   */
+  public BottomUpQuantification(UUID facilityId, UUID programId, UUID processingPeriodId) {
+    this.facilityId = facilityId;
+    this.programId = programId;
+    this.processingPeriodId = processingPeriodId;
+  }
+
+  /**
+   * Creates new instance based on data from the importer.
+   */
+  public static BottomUpQuantification newInstance(Importer importer) {
+    BottomUpQuantification buq = new BottomUpQuantification();
+
+    buq.setCreatedDate(importer.getCreatedDate());
+    buq.setModifiedDate(importer.getModifiedDate());
+    buq.setFacilityId(importer.getFacilityId());
+    buq.setProgramId(importer.getProgramId());
+    buq.setProcessingPeriodId(importer.getProcessingPeriodId());
+    buq.setTargetYear(importer.getTargetYear());
+    buq.setStatus(importer.getStatus());
+
+    return buq;
+  }
+
+  /**
+   * Copy values of attributes into new or updated BottomUpQuantification.
+   *
+   * @param original object to update from.
+   * @param lineItems list of bottom-up quantification line items.
+   */
+  public void updateFrom(BottomUpQuantification original,
+      List<BottomUpQuantificationLineItem> lineItems) {
+    setModifiedDate(ZonedDateTime.now());
+    setCreatedDate(original.getCreatedDate());
```

Comment:
I'm not sure, but I think the `created date` should never be updated.

Reply:
I removed **setCreatedDate(original.getCreatedDate());**.